runtime/doc/various.txt

```diff
@@ -557,5 +557,34 @@ LessInitFunc in your vimrc, for example: >
 	  set nocursorcolumn nocursorline
 	endfunc
 <
+==============================================================================
+3. Commenting							*commenting*
+
+Nvim supports commenting and uncommenting per line. It is done by toggling
+the status of consecutive lines: if all lines are commented - uncomment them,
+comment otherwise.
+
+Comment structure is directly inferred from 'commentstring' at cursor: text
+before and after `%s` is taken as left and right part of comment.
+
+Commenting line is done by inserting left comment part at appropriate indent
+and right part at line's end.
+Uncommenting is done by removing left and right parts from start and end of
+line (ignoring whitespace).
+Blank lines are also commented and uncommented with trimmed versions of
+comment parts for a more intuitive experience.
```

Comment:
I have to admit that I found this explanation not really helpful. Maybe consider expanding (explaining the structure of `commentstring`, giving an example) or just leaving this out as self-evident?

I think the most important part of the documentation is how it decides whether to comment, uncomment, or leave alone(?), so maybe just focus on that? (The specifics _how_ things are commented are not so important; I think it behaves exactly as people expect, modulo some edge cases which would be too tedious to document here.)

Reply:
I've replaced this paragraph with examples.